src/engagement_db_to_analysis/cache.py

```diff
@@ -48,38 +46,41 @@ def set_latest_message_timestamp(self, engagement_db_dataset, last_updated):
         :return: Latest run timestamp.
         :rtype: datetime.datetime
         """
-
         export_path = self._latest_message_timestamp_path(engagement_db_dataset)
         IOUtils.ensure_dirs_exist_for_file(export_path)
         with open(export_path, "w") as f:
             f.write(last_updated.isoformat())
 
-    def get_previous_export_messages(self, engagement_db_dataset):
+    def get_messages(self, engagement_db_dataset):
         """
-        Imports a list of messages for the given engagement_db_dataset from cache.
+        Gets a list of messages for the given engagement_db_dataset from the cache.
 
         :param engagement_db_dataset: Engagement db dataset name for this context.
         :type engagement_db_dataset: str
         :return: list of messages
-        :rtype: list
+        :rtype: list of engagement_database.data_models.Message
         """
+        previous_export_file_path = path.join(f"{self.cache_dir}/{engagement_db_dataset}.jsonl")
+        messages = []
+        try:
+            with open(previous_export_file_path) as f:
+                for line in f:
+                    messages.append(Message.from_dict(json.loads(line)))
+        except FileNotFoundError:
+            return []
 
-        previous_export_file_path = path.join(f"{self.cache_dir}/{engagement_db_dataset}.json")
-        with open(previous_export_file_path) as f:
-            previous_export = json.load(f)
-
-        return previous_export
+        return messages
 
-    def export_engagement_db_dataset(self, engagement_db_dataset, messages):
+    def set_messages(self, engagement_db_dataset, messages):
         """
-        Exports a list of messages for the given engagement_db_dataset to cache.
+        Sets a list of messages for the given engagement_db_dataset.
 
         :param engagement_db_dataset: Engagement db dataset name for this context.
         :type engagement_db_dataset: str
-        :return: list of messages
-        :rtype: list
+        :param messages: Messages to set.
```

Comment:
"Messages to set for the given engagement_db_dataset". I think this avoids thinking of type set

Reply:
updated